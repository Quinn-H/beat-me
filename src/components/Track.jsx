import React from 'react'

import Step from './Step'

const Track = (props) => {
  const track = props.track
  // console.log(track.steps[0])
  const steps = track.steps.map((step, index) => {
    return <Step key={index} step={step} />
  })
  return (
    <tr className="track">
      <th className="row">{track.name}</th>
      {steps}
    </tr>
  )
}

export default Track
