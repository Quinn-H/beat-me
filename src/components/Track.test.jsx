/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import Track from './Track'

test('Track renders correctly', () => {
  const name = "Kick"
  const steps = []
  const track = {name, steps}
  const tree = renderer.create(
    <Track track={track} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
