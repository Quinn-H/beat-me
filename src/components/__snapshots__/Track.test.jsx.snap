exports[`test Track renders correctly 1`] = `
<tr
  className="track">
  <th
    className="row">
    Kick
  </th>
</tr>
`;
